solar_consumer/data/fetch_gb_data.py

```diff
@@ -160,10 +165,46 @@ def fetch_gb_data_historic(regime: str) -> pd.DataFrame:
         gsp_yield_df["regime"] = regime
         gsp_yield_df["gsp_id"] = gsp_id
 
-        all_gsps_yields.append(gsp_yield_df)
+        if gsp_yield_df.empty:
+            missing_gsps.append(gsp_id)
+        else:
+            all_gsps_yields.append(gsp_yield_df)
+            if gsp_id == 0:
+                national_df = gsp_yield_df.copy()
 
         # TODO back up
         # if there is national but no gsps, make gsp from national
         # https://github.com/openclimatefix/solar-consumer/issues/105
 
+    if missing_gsps and national_df is not None:
```

Comment:
Rather than loading from the database, I think it would be better to load from PVLive api, this would make the code neater I think